cub/cub/device/dispatch/kernels/warpspeed/SpecialRegisters.cuh

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <cub/config.cuh>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cub/device/dispatch/kernels/warpspeed/makeWarpUniform.cuh> // makeWarpUniform.cuh
+
+#include <cuda/__ptx/instructions/get_sreg.h>
+#include <cuda/std/cstdint> // uint32_t
+
+CUB_NAMESPACE_BEGIN
+
+// Commonly used special registers that we should cache in registers or uniform
+// registers.
+struct SpecialRegisters
+{
+  const uint32_t clusterCtaRank;
+  const uint32_t blockIdxX;
+  const uint32_t threadIdxX;
+  const uint32_t warpIdx;
+  const uint32_t laneIdx;
+};
+
+[[nodiscard]] _CCCL_DEVICE_API inline SpecialRegisters getSpecialRegisters()
```

Comment:
Given that this is a globally useful feature we should consider moving it into libc++

Reply:
Note: the special registers here are just the ones that are useful at this point in time to warpspeed. There are a few other special registers, like `cga_cta_id` that are not (yet) included. If we move into libc++, we would need more testing and vetting. Also, we would have to make choices about whether to add special registers that are special registers on some architectures but not on others.

All in all: good idea, but I think we should defer this, as it requires a bit more design than the struct currently has.